src/els_indexer.erl

```diff
@@ -181,16 +181,10 @@ terminate(_, _State) ->
 try_index_file(FullName, SyncAsync) ->
   try
     Uri = els_uri:uri(FullName),
-    Extension = filename:extension(Uri),
-    case els_dt_document:is_extension_supported(Extension) of
-      true ->
-        lager:debug("Indexing file. [filename=~s]", [FullName]),
-        {ok, Text} = file:read_file(FullName),
-        Document   = els_dt_document:new(Uri, Text),
-        ok         = index_document(Document, SyncAsync);
-      false ->
-        lager:debug("Skipping file. [filename=~s]", [FullName])
-    end
+    lager:debug("Indexing file. [filename=~s]", [FullName]),
```

Comment:
Why is this change being reverted? I don't think it's a good idea to index all files which sit under the project, especially if we don't know what they are. Opening this files, reading the content and attempting to parse them (to crash) sounds like a waste. If we want to support other extensions, we should just include them in my opinion. What features from the server do we use for `other` files?

Reply:
During the initial indexing we are only processing files with extension `.hrl` and `.erl`, which will avoid indexing the bulk of all other types of files.

The crash is happening when we use the `els_dt_document:new/2` function, this happens in multiple places. I thought it would be more desirable to just handle the files that users mark as `erlang` type (i.e. in Emacs through specifying `erlang-mode`) than check the extension and type of file everytime we use the function above.

Reply:
Sorry, I missed the fact we have some additional filtering in the `index_dir` function. I thought we were trying to index all files.